File: app/src/main/java/com/kieran/winnipegbusbackend/enums/FavouritesListSortType.java
package com.kieran.winnipegbusbackend.enums;

public enum FavouritesListSortType {
    STOP_NUMBER_ASC,
    STOP_NUMBER_DESC,
    FREQUENCY_ASC,
    FREQUENCY_DESC;

    public static FavouritesListSortType getEnum(String value) {
        switch (Integer.parseInt(value)) {
            default: return STOP_NUMBER_ASC;
            case 1: return STOP_NUMBER_DESC;
            case 2: return FREQUENCY_ASC;
            case 3: return FREQUENCY_DESC;
        }
    }

}

File: app/src/main/java/com/kieran/winnipegbusbackend/enums/FavouritesListSortType.kt
package com.kieran.winnipegbusbackend.enums

enum class FavouritesListSortType {
    STOP_NUMBER_ASC,
    STOP_NUMBER_DESC,
    FREQUENCY_ASC,
    FREQUENCY_DESC;


    companion object {

        fun getEnum(value: String): FavouritesListSortType {
            return when (Integer.parseInt(value)) {
                1 -> STOP_NUMBER_DESC
                2 -> FREQUENCY_ASC
                3 -> FREQUENCY_DESC
                else -> STOP_NUMBER_ASC
            }
        }
    }

}
